src/DssSpell.sol

```diff
@@ -16,25 +16,108 @@
 
 pragma solidity 0.8.16;
 
-import "dss-exec-lib/DssExec.sol";
+// import "dss-exec-lib/DssExec.sol";
 import "dss-exec-lib/DssAction.sol";
+import { MCD, DssInstance } from "dss-test/MCD.sol";
 
-import { GemAbstract } from "dss-interfaces/ERC/GemAbstract.sol";
+import { UsdsInit } from "./dependencies/01-usds/UsdsInit.sol";
+import { UsdsInstance } from "./dependencies/01-usds/UsdsInstance.sol";
 
-interface ProxyLike {
-    function exec(address target, bytes calldata args) external payable returns (bytes memory out);
+import { SUsdsInit, SUsdsConfig } from "./dependencies/02-susds/SUsdsInit.sol";
+import { SUsdsInstance } from "./dependencies/02-susds/SUsdsInstance.sol";
+
+import { SkyInit } from "./dependencies/03-sky/SkyInit.sol";
+import { SkyInstance } from "./dependencies/03-sky/SkyInstance.sol";
+
+import { UniV2PoolMigratorInit } from "./dependencies/04-univ2-pool-migrator/UniV2PoolMigratorInit.sol";
+
+interface PauseLike {
+    function delay() external view returns (uint256);
+    function plot(address, bytes32, bytes calldata, uint256) external;
+    function exec(address, bytes32, bytes calldata, uint256) external returns (bytes memory);
+}
+
+interface ChainlogLike {
+    function getAddress(bytes32) external view returns (address);
+}
+
+interface SpellActionLike {
+    function officeHours() external view returns (bool);
+    function description() external view returns (string memory);
+    function nextCastTime(uint256) external view returns (uint256);
+}
+
+contract DssExec {
+    ChainlogLike  constant public chainlog = ChainlogLike(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+    uint256                public eta;
+    bytes                  public sig;
+    bool                   public done;
+    bytes32      immutable public tag;
+    address      immutable public action;
+    uint256      immutable public expiration;
+    PauseLike    immutable public pause;
+
+    uint256 constant public MIN_ETA = 1726574400; // 2024-09-17T12:00:00Z
```

Comment:
I think per checklist variable name should contain date

Reply:
Yeah, that's applicable to vests mostly... I don't think we have an established pattern for execution dates for spells.
Also, dates have either `T00:00:00Z` or `T:23:59:59Z`, unless we append the hour to the name, it would make no sense.

Reply:
> that's applicable to vests mostly...

I meant that there is a section in the checklist that enforces declaration of any timestamp:
https://github.com/makerdao/pe-checklists/blob/4b55876a9dec8b2c5ccaeb457fcbd134b629bb3a/spell/spell-reviewer-mainnet-checklist.md?plain=1#L75-L82

Reply:
Yes, still look at L80 for instance...
There's no "logical" sense here, the date is arbitrary.
We could use something like `SEP_08_2024_NOON_UTC` 🤷🏻‍♂️

Reply:
Still I believe calling this `MIN_ETA` is better to convey the meaning.
The checklists are thought to validate `DssSpellActions`, not custom `DssExec` implementations.

Reply:
> The checklists are thought to validate `DssSpellActions`, not custom `DssExec` implementations.

Not sure where this limitation is coming from. Checklists are there to enforce a certain quality and style of the code, I believe they should applied to anything related to the governance payload.